src/Mvc/Mvc.ApiExplorer/src/EndpointMethodInfoApiDescriptionProvider.cs

```diff
@@ -0,0 +1,213 @@
+                    foreach (var httpMethod in httpMethodMetadata.HttpMethods)
+                    {
+                        context.Results.Add(CreateApiDescription(routeEndpoint.RoutePattern, httpMethod, methodInfo));
+                    }
+                }
+            }
+        }
+
+        public void OnProvidersExecuted(ApiDescriptionProviderContext context)
+        {
+        }
+
+        internal static ApiDescription CreateApiDescription(RoutePattern pattern, string httpMethod, MethodInfo methodInfo)
+        {
+            var apiDescription = new ApiDescription
+            {
+                HttpMethod = httpMethod,
+                RelativePath = pattern.RawText?.TrimStart('/'),
+                ActionDescriptor = new ActionDescriptor
+                {
+                    RouteValues =
+                    {
+                        // Swagger uses this to group endpoints together.
+                        // For now, put all endpoints configured with Map(Delegate) together.
+                        // TODO: Use some other metadata for this.
+                        ["controller"] = "Map",
+                    },
+                },
+            };
+
+            var hasJsonBody = false;
+
+            foreach (var parameter in methodInfo.GetParameters())
+            {
+                var parameterDescription = CreateApiParameterDescription(parameter, pattern);
+
+                if (parameterDescription.Source == BindingSource.Body)
+                {
+                    hasJsonBody = true;
+                }
+
+                apiDescription.ParameterDescriptions.Add(CreateApiParameterDescription(parameter, pattern));
+            }
+
+            if (hasJsonBody)
+            {
+                apiDescription.SupportedRequestFormats.Add(new ApiRequestFormat
+                {
+                    MediaType = "application/json",
+                });
+            }
+
+            var responseType = methodInfo.ReturnType;
+
+            if (AwaitableInfo.IsTypeAwaitable(responseType, out var awaitableInfo))
+            {
+                responseType = awaitableInfo.ResultType;
+            }
+
+            responseType = Nullable.GetUnderlyingType(responseType) ?? responseType;
+
+            if (CreateApiResponseType(responseType) is { } apiResponseType)
+            {
+                apiDescription.SupportedResponseTypes.Add(apiResponseType);
+            }
+
+            return apiDescription;
+        }
+
+        private static ApiParameterDescription CreateApiParameterDescription(ParameterInfo parameter, RoutePattern pattern)
+        {
+            var parameterType = parameter.ParameterType;
+
+            var (source, name) = GetBindingSourceAndName(parameter, pattern);
+
+            return new ApiParameterDescription
+            {
+                Name = name,
+                ModelMetadata = new EndpointMethodInfoModelMetadata(ModelMetadataIdentity.ForType(parameterType)),
```

Comment:
Seems unfortunate we have to implement this. It doesn't look like there's anything other than the mdoel type that Swashbuckle reads from ModelMetadata.